### PR TITLE
[DOCS-1260] Add cookie_domain to Dev Portal session config example 

### PR DESCRIPTION
**Pull request by tlblessing**
https://konghq.atlassian.net/browse/DOCS-1260 Add "cookie_domain" info to "portal_session_conf" in Dev Portal docs

image

Added an example and links to the location where the cookie domain is discussed in detail. Also did a copy edit.

Direct review links:

Plugin hub

https://deploy-preview-2386--kongdocs.netlify.app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth/#enable-portal-session-config

https://deploy-preview-2386--kongdocs.netlify.app/enterprise/2.2.x/developer-portal/configuration/authentication/basic-auth/#enable-portal-session-config

https://deploy-preview-2386--kongdocs.netlify.app/enterprise/2.3.x/developer-portal/configuration/authentication/basic-auth/#enable-portal-session-config

Install topics (per Lena):

https://deploy-preview-2386--kongdocs.netlify.app/enterprise/2.1.x/deployment/installation/kong-on-kubernetes/#step-6-prepare-the-sessions-plugin-for-kong-manager-and-dev-portal

https://deploy-preview-2386--kongdocs.netlify.app/enterprise/2.2.x/deployment/installation/kong-on-kubernetes/#step-6-prepare-the-sessions-plugin-for-kong-manager-and-dev-portal

https://deploy-preview-2386--kongdocs.netlify.app/enterprise/2.3.x/deployment/installation/kong-on-kubernetes/#step-5-prepare-the-sessions-plugin-for-kong-manager-and-dev-portal

image

Note to self: This PR was currently updating a file I will also be updating: https://github.com/Kong/docs.konghq.com/pull/2391 so rebase after that PR is merged (done. twice.).

### CODE REVIEW

File: app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md [hunk 1]
```diff
@@ -2,56 +2,76 @@
...
+- [Kong configuration file](#enable-basic-auth-via-the-kong-conf)
+
+**Warnings:**
+
+- Enabling authentication in the Dev Portal requires use of the
+Sessions plugin. Developers will not be able to log in if this is not properly set.
+For more information, see
+[Sessions in the Dev Portal](/enterprise/{{page.kong_version}}/developer-portal/configuration/authentication/sessions).
+
+- When Dev Portal Authentication is enabled, content files remain unauthenticated until a role is applied to them. The exceptions are `settings.txt` and `dashboard.txt` that begin with the `*` role. For more information, see
```
Comment (lena-larionova) [inline app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md:23]:
```suggestion
- When Dev Portal Authentication is enabled, content files remain unauthenticated until a role is applied to them. The exceptions are `settings.txt` and `dashboard.txt`, which begin with the `*` role. For more information, see
```

File: app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md [hunk 2]
```diff
@@ -2,56 +2,76 @@
...
-> Sessions plugin. Developers will not be able to login if this is not set
-> properly. More information about [Sessions in the Dev Portal](/enterprise/{{page.kong_version}}/developer-portal/configuration/authentication/sessions)
 
 ### Enable Portal Session Config
 
 In the Kong configuration file, set the `portal_session_conf` property:
 
-```
-portal_session_conf={ "cookie_name": "portal_session", "secret": "CHANGE_THIS", "storage": "kong" }
+```bash
```
Comment (lena-larionova) [inline app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md:31]: This isn't really bash - you're editing `kong.conf`, which I guess you might do using bash, but you're just as likely to use a text editor. Same goes for the next two examples. 

I'd stick with no syntax highlighting here. `kong.conf` is a mix of formats - technically the latter part of the session config is in json, but the property name is not.

File: app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md [hunk 3]
```diff
@@ -2,56 +2,76 @@
...
 
->**Warning** When Dev Portal Authentication is enabled, content files will remain unauthenticated until a role is applied to them. The exception to this is `settings.txt` and `dashboard.txt` which begin with the `*` role. Please visit the <a href="/enterprise/{{page.kong_version}}/developer-portal/administration/developer-permissions">Developer Roles and Content Permissions</a> section for more info.
+![Authentication plugin](/assets/images/docs/dev-portal/portal-settings-auth-plugin.png)
 
-### Enable Basic Auth via the Command Line
+### Enable Basic Auth Using the Command Line
 
 To patch a Dev Portal's authentication property directly, run:
 
 ```
```
Comment (lena-larionova) [inline app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md:69]:
```suggestion
```bash
```

File: app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md [hunk 4]
```diff
@@ -2,56 +2,76 @@
...
+$ curl -X PATCH http://localhost:8001/workspaces/<WORKSPACE_NAME> \
   --data "config.portal_auth=basic-auth"
 ```
 
->**Warning** When Dev Portal Authentication is enabled, content files will remain unauthenticated until a role is applied to them. The exception to this is `settings.txt` and `dashboard.txt` which begin with the `*` role. Please visit the <a href="/enterprise/{{page.kong_version}}/developer-portal/administration/developer-permissions">Developer Roles and Content Permissions</a> section for more info.
-
-### Enable Basic Auth via the Kong.conf
+### Enable Basic Auth Using `kong.conf`
 
 Kong allows for a `default authentication plugin` to be set in the Kong
```
Comment (lena-larionova) [inline app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md:76]:
```suggestion
Kong allows for a default authentication plugin to be set in the Kong
```
Not really sure why that's in backticks, as it isn't the property name.

File: app/enterprise/2.1.x/developer-portal/configuration/authentication/sessions.md [hunk 5]
```diff
@@ -6,11 +6,11 @@ title: Sessions in the Dev Portal
 
 ## How does the Sessions Plugin work in the Dev Portal?
 
-When a user logs in to the Dev Portal with their credentials, the Sessions Plugin will create a session cookie. The cookie is used for all subsequent requests and is valid to authenticate the user. The session has a limited duration and renews at a configurable interval, which helps prevent an attacker from obtaining and using a stale cookie after the session has ended.
+When a user logs in to the Dev Portal with their credentials, the Sessions Plugin creates a session cookie. The cookie is used for all subsequent requests and is valid to authenticate the user. The session has a limited duration and renews at a configurable interval, which helps prevent an attacker from obtaining and using a stale cookie after a session has ended.
```
Comment (lena-larionova) [inline app/enterprise/2.1.x/developer-portal/configuration/authentication/sessions.md:9]:
```suggestion
When a user logs in to the Dev Portal with their credentials, the Sessions plugin creates a session cookie. The cookie is used for all subsequent requests and is valid to authenticate the user. The session has a limited duration and renews at a configurable interval, which helps prevent an attacker from obtaining and using a stale cookie after a session has ended.
```

File: app/enterprise/2.1.x/developer-portal/configuration/authentication/sessions.md [hunk 6]
```diff
@@ -28,21 +28,20 @@ portal_session_conf = {
 }
 ```
 
-* `"cookie_name":"<SET_COOKIE_NAME>"`: The name of the cookie
-  * For example, `"cookie_name":"portal_cookie"`
+* `"cookie_name":"<SET_COOKIE_NAME>"`: The name of the cookie. For example, `"cookie_name":"portal_cookie"`.
 * `"secret":"<SET_SECRET>"`: The secret used in keyed HMAC generation. Although
   the **Session Plugin's** default is a random string, the `secret` _must_ be
```
Comment (lena-larionova) [inline app/enterprise/2.1.x/developer-portal/configuration/authentication/sessions.md:33]:
```suggestion
  the **Session plugin's** default is a random string, the `secret` _must_ be
```

File: app/enterprise/2.1.x/developer-portal/configuration/authentication/sessions.md [hunk 7]
```diff
@@ -28,21 +28,20 @@ portal_session_conf = {
...
   manually set for use with the Dev Portal since it must be the same across all
   Kong workers/nodes.
 * `"storage":"kong"`: Where session data is stored. This value _must_ be set to `kong` for use with the Dev Portal.
-* `"cookie_lifetime":<NUMBER_OF_SECONDS_TO_LIVE>`: The duration (in seconds) that the session will remain open; 3600 by    default.
-* `"cookie_renew":<NUMBER_OF_SECONDS_LEFT_TO_RENEW>`: The duration (in seconds) of a session remaining at which point
-   the Plugin renews the session; 600 by default.
+* `"cookie_lifetime":<NUMBER_OF_SECONDS_TO_LIVE>`: The duration (in seconds) that the session will remain open; 3600 by default.
+* `"cookie_renew":<NUMBER_OF_SECONDS_LEFT_TO_RENEW>`: The duration (in seconds) of a session remaining at
+which point the Plugin renews the session; 600 by default.
```
Comment (lena-larionova) [inline app/enterprise/2.1.x/developer-portal/configuration/authentication/sessions.md:39]:
```suggestion
which point the plugin renews the session; 600 by default.
```

File: app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md [hunk 8]
```diff
@@ -156,6 +164,14 @@ In the following steps, replace `<your-password>` with a secure password.
     ```
     $ echo '{"cookie_name":"portal_session","cookie_samesite":"off","secret":"<your-password>","cookie_secure":false,"storage":"kong"}' > portal_session_conf
     ```
+
+    If you have different subdomains for the `portal_api_url` and `portal_gui_host`, set the `cookie_domain`
+    and `cookie_samesite` properties as follows:
+
+    ```
+    $ echo '{"cookie_name":"portal_session","cookie_samesite":"off", "cookie_domain":"<.your_subdomain.com">,"secret":"<your-password>","cookie_secure":false,"storage":"kong"}' > portal_session_conf
```
Comment (lena-larionova) [inline app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md:172]:
```suggestion
    $ echo '{"cookie_name":"portal_session","cookie_samesite":"off","cookie_domain":"<.your_subdomain.com">,"secret":"<your-password>","cookie_secure":false,"storage":"kong"}' > portal_session_conf
```

File: app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md [hunk 9]
```diff
@@ -156,6 +164,14 @@ In the following steps, replace `<your-password>` with a secure password.
     ```
     $ echo '{"cookie_name":"portal_session","cookie_samesite":"off","secret":"<your-password>","cookie_secure":false,"storage":"kong"}' > portal_session_conf
     ```
+
+    If you have different subdomains for the `portal_api_url` and `portal_gui_host`, set the `cookie_domain`
```
Comment (lena-larionova) [inline app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md:168]:
```suggestion
    Or, if you have different subdomains for the `portal_api_url` and `portal_gui_host`, set the `cookie_domain`
```
Putting that "or" here to hopefully stop people from doing this step twice.

File: app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md [hunk 10]
```diff
@@ -140,6 +140,14 @@ In the following steps, replace `<your-password>` with a secure password.
...
```
Comment (lena-larionova) [inline app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md:144]:
```suggestion
    Or, if you have different subdomains for the `portal_api_url` and `portal_gui_host`, set the `cookie_domain`
```

File: app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md [hunk 11]
```diff
@@ -140,6 +140,14 @@ In the following steps, replace `<your-password>` with a secure password.
...
+    and `cookie_samesite` properties as follows:
+
+    ```
+    $ echo '{"cookie_name":"portal_session","cookie_samesite":"off", "cookie_domain":"<.your_subdomain.com">,"secret":"<your-password>","cookie_secure":false,"storage":"kong"}' > portal_session_conf
```
Comment (lena-larionova) [inline app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md:148]:
```suggestion
    $ echo '{"cookie_name":"portal_session","cookie_samesite":"off","cookie_domain":"<.your_subdomain.com">,"secret":"<your-password>","cookie_secure":false,"storage":"kong"}' > portal_session_conf
```

**Review by lena-larionova — COMMENTED** [review]
For the topic "Enable Basic Auth in the Dev Portal", it might be cleaner to turn the latter three sections (methods for enabling auth) into three tabs instead of sections - kong manager, command line, kong.conf.

File: app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md [hunk 12]
```diff
@@ -2,56 +2,76 @@
...
-```
-portal_session_conf={ "cookie_name": "portal_session", "secret": "CHANGE_THIS", "storage": "kong" }
+```bash
+portal_session_conf={ "cookie_name":"portal_session","secret":"<CHANGE_THIS>","storage":"kong"}
 ```
 
 If using HTTP while testing, include `"cookie_secure": false` in the config:
 
+```bash
+portal_session_conf={ "cookie_name":"portal_session","secret":"<CHANGE_THIS>","storage":"kong","cookie_secure":"false"}
```
[the remark this answers is not in the document]
Comment (tlblessing) [reply app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md:38]: @lena-larionova I think line 35 is incorrect. I'll adjust. I updated that value to match your install docs. I thinks it all needs quotes. Need to test...

File: app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md (hunk 1 above)
Comment (tlblessing) [reply app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md:23]: stet. prefer less clauses where possible.

File: app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md (hunk 10 above)
Comment (tlblessing) [reply app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md:144]: stet. OR seems superfluous.

File: app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md (hunk 9 above)
Comment (tlblessing) [reply app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md:168]: stet.

File: app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (lena-larionova) [reply app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md:38]: For reference, see this guide on JSON datatypes: https://restfulapi.net/json-data-types/

Boolean values like this one do not require quotes.

**Comment by tlblessing** [comment]
Will need to port these updates to 2.2, 2.3, and soon 2.4. Hope we can version the repos and implement merge scripts soon.

File: app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md (hunk 10 above)
Comment (tlblessing) [reply app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md]: Done

File: app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md (hunk 2 above)
Comment (tlblessing) [reply app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md]: Done in commit eefd763

File: app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (tlblessing) [reply app/enterprise/2.1.x/developer-portal/configuration/authentication/basic-auth.md]: Yes, Sven's blog also shows no quotes for false https://svenwal.de/blog/20210316_kong_manager_install/

File: app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md (hunk 10 above)
Comment (tlblessing) [reply app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md]: Done in commit 5a41b6e

File: app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md (hunk 9 above)
Comment (tlblessing) [reply app/enterprise/2.1.x/kong-for-kubernetes/install-on-kubernetes.md]: Done

**Review by lena-larionova — APPROVED** [review]
LGTM, provided all of this is pulled into 2.2 and 2.3

**Pull request merged**